awx/main/models/credential/injectors.py

```diff
@@ -39,6 +39,20 @@ def gce(cred, env, private_data_dir):
     f.close()
     os.chmod(path, stat.S_IRUSR | stat.S_IWUSR)
     env['GCE_CREDENTIALS_FILE_PATH'] = path
+
+    handle, path = tempfile.mkstemp(dir=private_data_dir)
+    f = os.fdopen(handle, 'w')
+    json.dump(json_cred, f, indent=2)
+    f.close()
+    os.chmod(path, stat.S_IRUSR | stat.S_IWUSR)
+    env['GCP_SERVICE_ACCOUNT_FILE'] = path
+
+    # Handle env variables for new module types.
+    # This includes gcp_compute inventory plugin and
+    # all new gcp_* modules.
+    env['GCP_AUTH_KIND'] = 'serviceaccount'
+    env['GCP_PROJECT'] = project
+    env['GCP_ENV_TYPE'] = 'tower'
```

Comment:
Curious what is going on here. I can't find any documentation on this at a quick glance. I wonder if it is for future use. How might this be used?

Reply:
"GCP_AUTH_KIND" and "GCP_PROJECT" are part of the new set of modules (and the "gcp_compute" inventory plugin). Here's the [link](https://github.com/ansible/ansible/blob/a269c564c7cbdc76cd2d8bc3718e5e2e43b83a5f/lib/ansible/module_utils/gcp_utils.py#L190) to the AnsibleModule code.

"GCP_ENV_TYPE" is for future use. I figured it would be easier to add it here. The corresponding module PR is [here](https://github.com/modular-magician/ansible/pull/302). TLDR; it lets users change the user-agent string on API requests which lets them do more granular metrics.

Reply:
Thanks for those links, this is very helpful information to have. I might start documenting these on our side at some point.

If you could squash commits, that would help. I want to return to this, combine the cred files, and then patch the test to pass.

Reply:
Commits have been rebased!